File: Week1/TimeConversion/src/main/java/MyCode.java
package main.java;

public class MyCode {
    final static String DOCE = "12";
    final static String AM = "AM";
    final static String PM = "PM";

    public static String timeConversion(String s){
        String result = " ";
        if(s.contains(DOCE) && s.contains(PM)){
            result = "12:"+s.subSequence(3,8);
        } else if (s.contains(PM)) {
            int auxTime = (Integer.parseInt(s.subSequence(0,2).toString())+12 );
            result = String.format("%d:%s",auxTime, s.subSequence(3,8));
            
        } else if (s.contains(DOCE) && s.contains(AM)) {
            result = "00:"+s.subSequence(3,8);
            
        } else if (s.contains(AM)) {
            result = s.subSequence(0,8).toString();

        }
        return result;
    }
}

File: Week1/TimeConversion/src/main/kotlin/MyCode.kt
package main.kotlin

fun timeConversion(s:String):String{
    val format = s[8]
    val auxTime = s.subSequence(0, 2).toString()
    var result = " "
    if(format == 'P' && auxTime == "12"){
        result = "12:${s.subSequence(3, 8)}"
    }else if(format == 'P'){
        val time = auxTime.toInt() +12
        result = "$time:${s.subSequence(3, 8)}"
    }else if(format == 'A' && auxTime == "12"){
        result = "00:${s.subSequence(3, 8)}"

    }else if( format == 'A'){
        result = "${s.subSequence(0, 8)}"
    }
    return result

}
fun main(args: Array<String>) {
    val s = readLine()!!

    val result = timeConversion(s)

    println(result)
}